lib/jsdom/living/helpers/dates-and-times.js

```diff
@@ -212,7 +212,33 @@ function serializeWeek({ year, week }) {
   return `${yearStr}-W${weekStr}`;
 }
 
+// https://stackoverflow.com/a/6117889
+function parseDateAsWeek(originalDate) {
+  const dayInSeconds = 86400000;
+  // Copy date so don't modify original
+  const date = new Date(Date.UTC(originalDate.getUTCFullYear(), originalDate.getUTCMonth(), originalDate.getUTCDate()));
+  // Set to nearest Thursday: current date + 4 - current day number
+  // Make Sunday's day number 7
+  date.setUTCDate(date.getUTCDate() + 4 - (date.getUTCDay() || 7));
+  // Get first day of year
+  const yearStart = new Date(Date.UTC(date.getUTCFullYear(), 0, 1));
+  // Calculate full weeks to nearest Thursday
+  const week = Math.ceil((((date - yearStart) / dayInSeconds) + 1) / 7);
+
+  return { year: date.getUTCFullYear(), week };
+}
+
+function isDate(obj) {
+  try {
+    Date.prototype.valueOf.call(obj);
+    return true;
+  } catch {
```

Comment:
maybe should be compatible with Node.js 8.x.

Reply:
Node 8.x support has been dropped with version 16.0: https://github.com/jsdom/jsdom/blob/master/Changelog.md#1600